datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -901,15 +903,15 @@ struct ExprIdentifierVisitor<'a, 'n> {
     random_state: &'a RandomState,
     // a flag to indicate that common expression found
     found_common: bool,
+    // if we are in a conditional branch
```

Comment:
I think it would help to document more what is meant by 'conditional' means -- maybe like this

```suggestion
    // if we are in a conditional branch. A conditional
    // branch means that the expression **might** not be executed depending
    // on the runtime values of other expressions, and thus can not be extracted 
    // as a common expression . 
```

Reply:
Fixed in https://github.com/apache/datafusion/pull/11357/commits/b79a9a697566ea8ddadcf36f494b74a79826a526.